charts/trusted-artifact-signer/templates/cosign-deployment.yaml

```diff
@@ -12,27 +12,29 @@ spec:
       labels:
         app: {{ .Values.configs.cosign.name }}
     spec:
+      {{- if $.Values.configs.cosign.rolebindings }}
+      serviceAccountName: {{ index .Values.configs.cosign.rolebindings 0 }}
+      {{- end }}
       containers:
       - name: {{ .Values.configs.cosign.name }}
         image: "{{ .Values.configs.cosign.image.registry }}/{{ .Values.configs.cosign.image.repository }}:{{ .Values.configs.cosign.image.version }}"
         env:
         - name: OPENSHIFT_APPS_SUBDOMAIN
           value: {{ .Values.configs.cosign.appsSubdomain }}
-        - name: KEYCLOAK_REALM
+        - name: OIDC_AUTHENTICATION_REALM
           value: "sigstore"
         - name: FULCIO_URL
           value: "https://fulcio.$(OPENSHIFT_APPS_SUBDOMAIN)"
-        - name: KEYCLOAK_URL
-          value: "https://keycloak-keycloak-system.$(OPENSHIFT_APPS_SUBDOMAIN)"
+        - name: OIDC_ISSUER_URL
+          value: "https://keycloak-keycloak-system.$(OPENSHIFT_APPS_SUBDOMAIN)/auth/realms/$(OIDC_AUTHENTICATION_REALM)"
         - name: REKOR_URL
           value: "https://rekor.$(OPENSHIFT_APPS_SUBDOMAIN)"
         - name: TUF_URL
           value: "https://tuf.$(OPENSHIFT_APPS_SUBDOMAIN)"
-        - name: KEYCLOAK_OIDC_ISSUER
-          value: "$KEYCLOAK_URL/auth/realms/$(KEYCLOAK_REALM)"
         securityContext:
```

Comment:
@sallyom when testing the cosign changes, I noticed that although not creating an error, the cosign deployment was not properly being deployed you can find this msg in the yaml file of the cosign deployment:

```
      reason: FailedCreate
      message: >-
        pods "cosign-96df6bd59-" is forbidden: unable to validate against any
        security context constraint:
        [pod.metadata.annotations[container.seccomp.security.alpha.kubernetes.io/cosign]:
        Forbidden: seccomp may not be set, provider restricted-v2:
        .containers[0].runAsUser: Invalid value: 65533: must be in the ranges:
        [1001350000, 1001359999], provider "restricted": Forbidden: not usable
        by user or serviceaccount, provider "nonroot-v2": Forbidden: not usable
        by user or serviceaccount, provider "nonroot": Forbidden: not usable by
        user or serviceaccount, provider "pcap-dedicated-admins": Forbidden: not
        usable by user or serviceaccount, provider "hostmount-anyuid":
        Forbidden: not usable by user or serviceaccount, provider
        "hostnetwork-v2": Forbidden: not usable by user or serviceaccount,
        provider "hostnetwork": Forbidden: not usable by user or serviceaccount,
        provider "hostaccess": Forbidden: not usable by user or serviceaccount,
        provider "node-exporter": Forbidden: not usable by user or
        serviceaccount, provider "privileged": Forbidden: not usable by user or
        serviceaccount]
```

I believe this is because the securityContext is not being set properly as its inside the containers array, and should be in the spec object, also somethings cant be set here so they need to be removed.

```
    spec:
      containers:
      securityContext:
        runAsNonRoot: true
        runAsUser: 65533
```

Reply:
hm... ok i'll update that, thank you!

Reply:
yea I thought it was weird too alright

Reply:
Really sorry @sallyom , looking at the way I made my comment it looks confusing, 

```
      securityContext:
        runAsNonRoot: true
        runAsUser: 65533
```

These are the fields that should be kept in, again apologies

Reply:
oh! :eyes: so there's 
```yaml
spec:
  securityContext:
    outside-container-spec
   containers:
     securityContext:
       inside-container-spec
```
so which lines should be where? 
I believe all of this info should be somewhere:
```
allowPrivilegeEscalation: false
runAsNonRoot: true
capabilities:
  drop:
  - ALL
runAsNonRoot: true
runAsUser: 65533
```
so which goes where? i'm confused - as/is things are working in Kind - but probably you're seeing fails in OpenShift, right?